### PR TITLE
gpu/amd: drop hardware.amdgpu.opencl option

### PR DESCRIPTION
**Pull request by Mic92**
This now conflicts with nixpkgs. See https://nixpk.gs/pr-tracker.html?pr=319865

You will need to update from `hardware.amdgpu.opencl = true;` to `hardware.amdgpu.opencl.enable = true;`

###### Description of changes

See https://github.com/NixOS/nixpkgs/pull/319865

The other options we will migrate too, once they have landed in stable/unstable channels

###### Things done

- [ ] Tested the changes in your own NixOS Configuration
- [ ] Tested the changes end-to-end by using your fork of `nixos-hardware` and
      importing it via `<nixos-hardware>` or Flake input

### CODE REVIEW

**Comment by Mic92** [comment]
A few people seem to use this:
https://github.com/search?q=hardware.amdgpu.opencl+lang%3Anix&type=code

cc @blitz @necrophcodr @jocelynthode @felipecaninnovaes @jpaali @nightconcept @EnzioKam

**Comment by Mic92** [comment]
You will need to update nixpkgs to the version that has this option or pin nixos-hardware to the old version.
@nevivurn @tupakkatapa @DrymarchonShaun @arvigeus @anna328p @Mange @wochap

**Comment by Mic92** [comment]
cc @jzbor @CallumTarttelin @knightpp @spagnoloG

File: common/gpu/amd/default.nix [hunk 1]
```diff
@@ -6,11 +6,6 @@
   ) // {
     default = true;
   };
-  options.hardware.amdgpu.opencl = lib.mkEnableOption (lib.mdDoc
-    "rocm opencl runtime (Install rocmPackages.clr and rocmPackages.clr.icd)"
-  ) // {
-    default = true;
```
Comment (Mic92) [inline common/gpu/amd/default.nix:12]: Once we have the option, we maybe should default this to true?

**Comment by Mic92** [comment]
An alternative hotfix is to just import the packages yourself:

```
{
  hardware.opengl.extraPackages = with pkgs.rocmPackages; [ clr clr.icd ];
}
```

**Comment by Mic92** [comment]
@mergify queue

File: common/gpu/amd/default.nix (hunk 1 above)
Comment (JohnRTitor) [reply common/gpu/amd/default.nix:12]: For this module? Yeah I think it should use 
```nix
hardware.amdgpu.rocm.enable = lib.mkDefault true;
```

Comment (Mic92) [reply common/gpu/amd/default.nix:12]: Wasn't this called `hardware.amdgpu.opencl.enable = lib.mkDefault true;` now?

**Comment by Mic92** [comment]
@mergify queue

File: common/gpu/amd/default.nix (hunk 1 above)
Comment (JohnRTitor) [reply common/gpu/amd/default.nix:12]: yeah, sorry, that was a mistake on my part.

**Comment by Mic92** [comment]
@mergify dequeue

**Comment by Mic92** [comment]
@mergify queue

**Comment by blitz** [comment]
So this means that nixos-hardware master will not be compatible with 24.05 wrt to amdgpu settings? If so would it make sense to just create a branch/tag for the last working version?

**Comment by Mic92** [comment]
> So this means that nixos-hardware master will not be compatible with 24.05 wrt to amdgpu settings? If so would it make sense to just create a branch/tag for the last working version?

The module has been backported to 24.05, so branching won't change anything here.

**Pull request merged**